### PR TITLE
Add Nebari local deployment instructions

### PR DESCRIPTION
**Pull request by pavithraes**
## Reference Issues or PRs

Partially addresses: #108 

Supersedes a portion of #113

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [x] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [x] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [x] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [x] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

## Any other comments?

**While squash merging, add @viniciusdc as co-author on this PR.**

### CODE REVIEW

**Review by viniciusdc — COMMENTED** [review]
Thanks, @pavithraes, I added a few suggestions in an attempt to provide more distinction between what `local` means and why should a user use it. Not sure if its really helpful, regarding the "existing" method. You can add something as: 


"""
If you are looking for installation and deployment instructions for an existing Kubernetes cluster, please visit [How to install and setup Nebari on an existing Kubernetes infrastructure](/how-tos/nebari-kubernetes).
"""

**Comment by pavithraes** [comment]
@viniciusdc Thank you for taking a look, and for all your suggestions, I've accepted them! If this looks good to you, we can merge. :)

File: docs/docs/how-tos/nebari-local.md
Comment (iameskild) [inline docs/docs/how-tos/nebari-local.md:34]: Unfortunately `local` MacOS deployment don't work at the moment. I was working on this a few months ago, see my notes in [this issue](https://github.com/nebari-dev/nebari/issues/1405).

File: docs/docs/how-tos/nebari-local.md
Comment (iameskild) [inline docs/docs/how-tos/nebari-local.md:163]: If I recall correctly, the SSL certs don't work for local deployments. 

Can you confirm this @viniciusdc?

**Review by iameskild — COMMENTED** [review]
Thanks for working on this @pavithraes! I made a few minor comments that we might want to sort out before merging otherwise this is looking good!

File: docs/docs/how-tos/nebari-local.md
Comment (viniciusdc) [reply docs/docs/how-tos/nebari-local.md:163]: Uhm, I would need to re-validate that info with a local install to make sure as well... but if I record correctly from previous deployments, I know that both Traefik/Let's Encrypts associated certs works just fine. Assuming in the case of Traefik that works means having to type `thisisunsafe`

File: docs/docs/how-tos/nebari-local.md
Comment (pavithraes) [reply docs/docs/how-tos/nebari-local.md]: Thanks for the note,  updated!

File: docs/docs/how-tos/nebari-local.md
Comment (pavithraes) [reply docs/docs/how-tos/nebari-local.md:163]: I've opened #243 to track this discussion to unblock this PR. :)

**Review by iameskild — APPROVED** [review]
Thanks @pavithraes 💯

**Review by viniciusdc — APPROVED** [review]

**Pull request merged**
